### ISSUE TITLE
Question about Aux Rules (again)

### ISSUE THREAD

**Issue by eurobank** [CIDRAM/CIDRAM#504]
See this:

ID: 1692737854-988984-9412203451 »
Date/Time: Tue, 22 Aug 2023 23:57:34 +0300 »
IP address: 8.222.179.80 »
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36 »
Signatures reference: abuseipdb.php:L143 »
Why blocked: AbuseIPDB Lookup! »
Reconstructed URI: https: // removed . com/contact/ »
Infractions: 1 »
Country code lookup: SG »
Request method: GET »
Hostname: - »

This ASN is also blocked with an Aux Rule. But no check was done, only AbuseIPDB.

Is this by design?

**Comment by eurobank** [comment]
Ok, i see that different ASNs are reported,

[AS45102](https://ipinfo.io/AS45102) - Alibaba (US) Technology Co., Ltd.

AS number	[134963](https://cleantalk.org/blacklists/as134963

so it is ok.

**Comment by Maikuolan** [comment]
> Is this by design?

Nope. There are *some modules* which, by design, won't perform any lookups or additional checks if it's detected that there's already a positive signatures count for the request at hand (e.g., by the default signature files), in order to minimise running up API quotas needlessly and to improve overall performance. However, whether auxiliary rules are executed for the request at hand shouldn't be affected at all by the signatures count (i.e., as long as the execution stage for auxiliary rules is enabled, and some auxiliary rules exist, they should execute). That's because for many users, auxiliary rules will be absolutely necessary for enforcing custom bypasses that they may need for countering false positives or other problems at their particular installation, caused by existing signature files or modules.

The one exception to that is when whitelisting has been triggered for the request at hand by the point the execution stage for auxiliary rules begins (e.g., if a whitelist signature in a custom signature file has been triggered, then that, by design, could prevent auxiliary rules from executing, per the same principle of improving overall performance, in that there's not much point doing any further processing the moment the decision is made to definitively not block something). However, the fact that the request was blocked at all is enough to show that that exception isn't in effect here.

In other words: If an auxiliary rule exists which should've blocked that request, and it failed to do so, then something's wrong somewhere, whether that be a bug in CIDRAM's code somewhere, a bug in the API responsible for fetching the ASN, a bug in the auxiliary rule itself, or something else. Not executing at all is not expected behaviour, in any case.

Offhandedly, I'm not sure the exact cause of the problem. I'll experiment a little at my side later today, to see whether I can figure it out.

**Comment by eurobank** [comment]
This is happening a lot. While some kind of block is in place, the IP is not blocked. 

For example:

Date/Time: Fri, 29 Sep 2023 16:23:05 +0300 »
IP address: 5.227.24.133 »
Query: action=register »
Referrer: https : // www . xxxxx . xx /wp-login.php?action=register »
User agent: Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36 »
Signatures reference: abuseipdb.php:L143 »
Why blocked: AbuseIPDB Lookup! »
Reconstructed URI: https : // www . xxxxxxxxx . xx/**wp-login.php?action=register** »
Infractions: 1 »
Country code lookup: RU »
Request method: GET »
Hostname: - »

This should be blocked

BUT i see it incoming in live view of Wordpress firewall.

[29/Sep/23:16:23:46 +0300] - 5.227.24.133 "[29/Sep/23:16:23:06 +0300] - 5.227.24.133 "GET /" ["https: / / xxx/**wp-login.php?action=register**"] "Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36" "-""

I have seen many many issues like that. We could say that the lookup of AbuseIPDB failed, but it is not as you can see. Same issues with ASNs.

I think you should check this at some point.
